AgileCoffeeShop/Program.cs

```diff
@@ -0,0 +1,130 @@
+                    coffeeShop.Name = splitLine[0];
+                    ValidateCoordinate(splitLine[1]);
+                    coffeeShop.Coordinate.YCoordinate = Convert.ToDouble(splitLine[1]);
+                    ValidateCoordinate(splitLine[2]);
+                    coffeeShop.Coordinate.XCoordinate = Convert.ToDouble(splitLine[2]);
+                    coffeeShops.Add(coffeeShop);
+                }
+            }
+            catch(Exception exception)
+            {
+                Console.WriteLine("Invalid URL." + exception.Message + "Program will exit.\n Press any key to continue...");
+                Console.ReadKey();
+            }
+        }
+
+        public static void ValidateCoordinate (string coordinate)
+        {
+            double temp = 0;
+            try
+            {
+                temp = Convert.ToDouble(coordinate);
+            }
+            catch(Exception exception)
+            {
+                Console.WriteLine("Invalid coordinates." + exception.Message + "Program will exit.\nPress any key to continue...");
+                Console.ReadKey();
+            }
+        }
+        private static void ValidateUrl(string url) 
+        {
+            bool result = Uri.TryCreate(url, UriKind.Absolute, out Uri uriResult) && (uriResult.Scheme == Uri.UriSchemeHttp || uriResult.Scheme == Uri.UriSchemeHttps);
+            if(result==true)
+            {
+                if(!url.Contains("raw.githubusercontent.com"))
+                {
+                    result = false;
+                }
+            }
+            if(result==false)
+            {
+                Console.WriteLine("Invalid URL. Program will exit.\n Press any key to continue...");
+                Console.ReadKey();
+            }
+        }
+
+        private static void Init()
+        {
+            user = new User();
+            urlFilePath = string.Empty;
+            coffeeShops = new List<CoffeeShop>();
+            distancesDictionary = new Dictionary<string, double>();
+        }
+
+        public static double CalculateDistance(Coordinate A, Coordinate B)
+        {
+            var distance = Math.Round(Math.Sqrt(Math.Pow((A.XCoordinate - B.XCoordinate), 2) + Math.Pow((A.YCoordinate - B.YCoordinate), 2)), 4);
+            return distance;
+        }
+
+        private static void CreateDistancesDictionary()
+        {
+            foreach (CoffeeShop coffeeShop in coffeeShops)
+            {
+                var distance = CalculateDistance(user.Coordinate, coffeeShop.Coordinate);
+                distancesDictionary.Add(coffeeShop.Name, distance);                
+            }
+            distancesDictionary = distancesDictionary.OrderBy(x => x.Value).ToDictionary(x => x.Key, x => x.Value);
+        }
+
+        private static void DisplayTheClosestCoffeeShops()
+        {
+            int counter = 0;
+
+            foreach (KeyValuePair<string, double> kvp in distancesDictionary)
+            {
+                if (counter < 3)
```

Comment:
Why are you using this magic value 3? 
Who is 3? Why using 3?

Reply:
I realized I was wrong here and I will use a constant int SHOPS_TO_DISPLAY.